packages/lesswrong/server/authenticationMiddlewares.ts

```diff
@@ -106,29 +151,33 @@ export const addAuthMiddlewares = (addConnectHandler) => {
       callbackURL: `${getSiteUrl()}auth/google/callback`,
       proxy: true
     },
-    createOAuthUserHandler('services.google.id', profile => profile.id, async profile => ({
-      email: profile.emails[0].address,
+    createOAuthUserHandler<GoogleProfile>('services.google.id', profile => profile.id, async profile => ({
+      email: profile.emails?.[0].value,
       services: {
         google: profile
       },
       emails: profile.emails,
       username: await Utils.getUnusedSlugByCollectionName("Users", slugify(profile.displayName)),
       displayName: profile.displayName,
       emailSubscribedToCurated: true
-    }))
+      // Type assertion here is because @types/passport-google-oauth20 doesn't
+      // think their verify callback is able to take a null in the place of the
+      // error, which seems like a bug and which prevents are seemingly working
+      // code from type-checking
+    })) as (_accessToken: string, _refreshToken: string, profile: GoogleProfile, done: GoogleVerifyCallback) => Promise<void>
   ))}
   
   const facebookClientId = facebookClientIdSetting.get()
-  if (facebookClientId) {
+  const facebookOAuthSecret = facebookOAuthSecretSetting.get()
+  if (facebookClientId && facebookOAuthSecret) {
     passport.use(new FacebookOAuthStrategy({
       clientID: facebookClientId,
-      clientSecret: facebookOAuthSecretSetting.get(),
+      clientSecret: facebookOAuthSecret,
       callbackURL: `${getSiteUrl()}auth/facebook/callback`,
       profileFields: ['id', 'emails', 'name', 'displayName'],
-      proxy: true
```

Comment:
In spite of what DefinitelyTyped thinks, this is a real option; looking at the source of `passport-oauth2`, it seems to affect whether the `x-forwarded-host` http header is used. But... I have no reason to think that matters, this option probably tagged along in a copy-paste from sample code? I think it's fine to take out, but needs to be tested in the actual server environment (local testing doesn't apply). (That'll be on me to test after this merges to devel.)